### PR TITLE
Add document about choosing the right scheduler

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/distributed/issues/364

### CODE REVIEW

File: docs/source/scheduler-choice.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+===========================
+
+Dask enables you to run your computations on a variety of schedulers that use
+different technologies, ranging from a single thread, to multiple processes, to
+a distributed cluster.  This document helps explain the differences between the
+different choices so that users can make decisions that improve performance.
+
+Briefly, the current options are as follows:
+
+*   ``dask.threaded.get``: Use multiple threads in the same process.  Good for
```
Comment (jcrist) [inline docs/source/scheduler-choice.rst]: "Use" -> "Uses"

File: docs/source/scheduler-choice.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    numeric code that releases the GIL (NumPy, Pandas, SKLearn, Numba) because
+    data is free to share.  The default on dask.array and dask.dataframe and
+    dask.delayed
```
Comment (jcrist) [inline docs/source/scheduler-choice.rst]: Perhaps:

"The default scheduler for ``dask.array``, ``dask.dataframe``, and ``dask.delayed``."

File: docs/source/scheduler-choice.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+*   ``dask.multiprocessing.get``: Uses multiple processes.  Good for Python
+    bound code that needs multiple interpreters to accelerate.  Some costs to
```
Comment (jcrist) [inline docs/source/scheduler-choice.rst]: Perhaps "There are some costs ..."

File: docs/source/scheduler-choice.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    data sharing back and forth between processes.  The default on dask.bag and
```
Comment (jcrist) [inline docs/source/scheduler-choice.rst]: "The default on" -> "The default scheduler for"?

**Comment by jcrist** [comment]
Besides my small nitpicks, this looks good to me.

File: docs/source/scheduler-choice.rst (hunk 4 above)
Comment (mrocklin) [inline docs/source/scheduler-choice.rst]: Fixed

File: docs/source/scheduler-choice.rst (hunk 3 above)
Comment (mrocklin) [inline docs/source/scheduler-choice.rst]: fixed

File: docs/source/scheduler-choice.rst (hunk 2 above)
Comment (mrocklin) [inline docs/source/scheduler-choice.rst]: Fixed

File: docs/source/scheduler-choice.rst (hunk 1 above)
Comment (mrocklin) [inline docs/source/scheduler-choice.rst]: Fixed

**Pull request merged**